bigbluebutton-html5/imports/ui/components/presentation/component.jsx

```diff
@@ -1,78 +1,246 @@
   }
 
+  componentDidMount() {
+    // adding an event listener to scale the whiteboard on 'resize' events sent by chat/userlist etc
+    window.addEventListener('resize', () => {
+      setTimeout(this.handleResize.bind(this), 0);
+    });
+
+    this.getInitialPaperSizes();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', () => {
+      setTimeout(this.handleResize.bind(this), 0);
+    });
+  }
+
+  // returns a ref to the svg element, which is required by a WhiteboardOverlay
+  // to transform screen coordinates to svg coordinate system
+  getSvgRef() {
+    return this.svggroup;
+  }
+
+  getPaperSizes() {
+    const { presentationPaper, whiteboardSizeAvailable } = this;
+    const paperSizes = {};
+
+    if (presentationPaper) {
+      // if a user is a presenter - this means there is a whiteboardToolBar on the right
+      // and we have to get the width/height of the whiteboardSizeAvailable
+      // (inner hidden div with absolute position)
+      let clientHeight;
+      let clientWidth;
+      if (this.props.userIsPresenter) {
+        clientHeight = whiteboardSizeAvailable.clientHeight;
+        clientWidth = whiteboardSizeAvailable.clientWidth;
+      // user is not a presenter - we can get the sizes of the presentationPaper
+      // direct parent of the svg wrapper
+      } else {
+        clientHeight = presentationPaper.clientHeight;
+        clientWidth = presentationPaper.clientWidth;
+      }
+
+      paperSizes.paperHeight = clientHeight;
+      paperSizes.paperWidth = clientWidth;
+    }
+    return paperSizes;
+  }
+
+  getInitialPaperSizes() {
+    // determining the paperWidth and paperHeight (available space for the svg) on the initial load
+    const paperSizes = this.getPaperSizes();
+    if (Object.keys(paperSizes).length > 0) {
+      // setting the state of the paperWidth and paperHeight (available space for the svg)
+      // and set the showSlide to true to start rendering the slide
+      paperSizes.showSlide = true;
+      this.setState(paperSizes);
+    }
+  }
+
+  handleResize() {
+    const paperSizes = this.getPaperSizes();
+    if (Object.keys(paperSizes).length > 0) {
+      // updating the size of the space available for the slide
+      this.setState(paperSizes);
+    }
+  }
+
+  calculateSize() {
+    const originalWidth = this.props.currentSlide.width;
+    const originalHeight = this.props.currentSlide.height;
+
+    let adjustedWidth;
+    let adjustedHeight;
+
+    // Slide has a portrait orientation
+    if (originalWidth <= originalHeight) {
+      adjustedWidth = (this.state.paperHeight * originalWidth) / originalHeight;
+      if (this.state.paperWidth < adjustedWidth) {
+        adjustedHeight = (this.state.paperHeight * this.state.paperWidth) / adjustedWidth;
+        adjustedWidth = this.state.paperWidth;
+      } else {
+        adjustedHeight = this.state.paperHeight;
+      }
+
+      // Slide has a landscape orientation
+    } else {
```

Comment:
What do you mean?
The slide is either portrait or landscape, I can only remove `else` if I add a `return` in the `if` block. Not sure how that would be better though.

Reply:
I suggest that we leave this case.
Soon it will be something like:
```
if(portrait)
   if(fit-to-width)
   ...
   else fit-to-page
else landscape
   if(fit-to-width)
   ...
   else fit-to-page
```

Having 4 early exits here will make it more difficult to read.